jerry-core/ecma/operations/ecma-arraybuffer-object.c

```diff
@@ -268,6 +273,142 @@ ecma_arraybuffer_detach (ecma_object_t *object_p) /**< pointer to the ArrayBuffe
   return true;
 } /* ecma_arraybuffer_detach */
 
+ecma_value_t
+ecma_builtin_arraybuffer_slice (ecma_value_t this_arg,
+                                const ecma_value_t *argument_list_p,
+                                uint32_t arguments_number)
+{
+  ecma_object_t *object_p = ecma_get_object_from_value (this_arg);
+
+  /* 4. */
+  if (ecma_arraybuffer_is_detached (object_p))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG (ecma_error_arraybuffer_is_detached));
+  }
+
+  /* 5. */
+  uint32_t len = ecma_arraybuffer_get_length (object_p);
+
+  uint32_t start = 0;
+  uint32_t end = len;
+
+  if (arguments_number > 0)
+  {
+    /* 6-7. */
+    if (ECMA_IS_VALUE_ERROR (ecma_builtin_helper_uint32_index_normalize (argument_list_p[0],
+                                                                               len,
+                                                                               &start)))
+    {
+      return ECMA_VALUE_ERROR;
+    }
+
+    if (arguments_number > 1 && !ecma_is_value_undefined (argument_list_p[1]))
+    {
+      /* 8-9. */
+      if (ECMA_IS_VALUE_ERROR (ecma_builtin_helper_uint32_index_normalize (argument_list_p[1],
+                                                                                 len,
+                                                                                 &end)))
+      {
+        return ECMA_VALUE_ERROR;
+      }
+    }
+  }
+
+  /* 10. */
+  uint32_t new_len = (end >= start) ? (end - start) : 0;
+
+  /* 11. */
+  ecma_value_t ctor;
+  if (ecma_is_shared_arraybuffer (this_arg))
+  {
+    ctor = ecma_op_species_constructor (object_p, ECMA_BUILTIN_ID_SHARED_ARRAYBUFFER);
+  }
+  else
+  {
+    ctor = ecma_op_species_constructor (object_p, ECMA_BUILTIN_ID_ARRAYBUFFER);
+  }
+
+  if (ECMA_IS_VALUE_ERROR (ctor))
+  {
+    return ctor;
+  }
+
+  /* 12. */
+  ecma_object_t *ctor_obj_p = ecma_get_object_from_value (ctor);
+  ecma_value_t new_len_value = ecma_make_uint32_value (new_len);
+
+  ecma_value_t new_arraybuffer = ecma_op_function_construct (ctor_obj_p, ctor_obj_p, &new_len_value, 1);
+
+  ecma_deref_object (ctor_obj_p);
+  ecma_free_value (new_len_value);
+
+  if (ECMA_IS_VALUE_ERROR (new_arraybuffer))
+  {
+    return new_arraybuffer;
+  }
+
+  ecma_object_t *new_arraybuffer_p = ecma_get_object_from_value (new_arraybuffer);
+  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+
+  /* 13. */
+  if (!(ecma_object_class_is (new_arraybuffer_p, ECMA_OBJECT_CLASS_ARRAY_BUFFER)
+        || ecma_object_class_is (new_arraybuffer_p, ECMA_OBJECT_CLASS_SHARED_ARRAY_BUFFER)))
+  {
+    ret_value = ecma_raise_type_error (ECMA_ERR_MSG ("Return value is not an ArrayBuffer object"));
+    goto free_new_arraybuffer;
+  }
+
+  /* 14-15. */
+  if (ecma_arraybuffer_is_detached (new_arraybuffer_p))
+  {
+    ret_value = ecma_raise_type_error (ECMA_ERR_MSG ("Returned ArrayBuffer has been detached"));
+    goto free_new_arraybuffer;
+  }
+
+  /* 16. */
+  if (ecma_op_same_value (new_arraybuffer, this_arg))
```

Comment:
since `new_arraybuffer` is an object, no `op_same_value` is required. Simply use `==`